chasm/lib/activity/frontend.go

```diff
@@ -85,6 +85,24 @@ func (h *frontendHandler) StartActivityExecution(ctx context.Context, req *workf
 	return resp.GetFrontendResponse(), err
 }
 
+// PollActivityExecution handles PollActivityExecutionRequest. This method supports querying current
+// activity state, optionally as a long-poll that waits for certain state changes. It is used by
+// clients to poll for activity state and/or result.
+func (h *frontendHandler) PollActivityExecution(
```

Comment:
You're missing validation here. At minimum, we'll need to validate that the activity ID is not empty and a check that run ID is provided if long poll token is provided.

Reply:
Are there flag combinations that are invalid maybe?

Reply:
The PR performs input validation now. I don't think there are invalid combinations other than the rule you said: token => runID